packages/gatsby/src/schema/schema.js

```diff
@@ -245,7 +245,7 @@ const createTypeComposerFromGatsbyType = ({
       return InterfaceTypeComposer.createTemp(type.config, schemaComposer)
     }
     default: {
-      console.warn(`Illegal type definition: ${JSON.stringify(type.config)}`)
+      report.warn(`Illegal type definition: ${JSON.stringify(type.config)}`)
```

Comment:
if you replace it here. Maybe also replace it in this line to have all on `report`:

`// console.log(printSchema(schema))`